abstract-factory/src/main/java/com/iluwatar/abstractfactory/App.java

```diff
@@ -41,12 +44,19 @@
  * and its implementations ( {@link ElfKingdomFactory}, {@link OrcKingdomFactory}). The example uses
  * both concrete implementations to create a king, a castle and an army.
  */
+@Slf4j
 public class App {
 
-  private static final Logger LOGGER = LoggerFactory.getLogger(App.class);
-
+  @Setter
```

Comment:
You can use `@Data` instead of adding getter and setter to each field

Reply:
Yes, I can, but it seemed to me that the annotation `@Data` is not quite suitable for the class `App`. Usually, the classes called `App` are needed to run an application. When I look at this class, I get the feeling that there is another class called `Kingdom'. I propose to move the fields `king`, `castle`, `army` and static class `FactoryMaker` from class `App to a new class `Kingdom` (where I can use the annotation `@Data`) and in `App to use an object of class `Kingdom`. This option seems cleaner to me. What do you think about it?

Reply:
Yeah you can do that as well